src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -171,34 +175,39 @@ private void setArgvGlobals(StringNodes.MakeStringNode makeStringNode) {
             }
         }
 
-        private Source loadMainSourceSettingDollarZero(
-                CallDispatchHeadNode findSFile,
-                StringNodes.MakeStringNode makeStringNode) {
+        private Source loadMainSourceSettingDollarZero(CallDispatchHeadNode findSFile, StringNodes.MakeStringNode makeStringNode, ExecutionAction executionAction, String toExecute) {
             final Source source;
             final Object dollarZeroValue;
-
             try {
-                final String to_execute = getContext().getOptions().TO_EXECUTE.intern();
-                switch (getContext().getOptions().EXECUTION_ACTION) {
+                switch (executionAction) {
                     case UNSET:
-                        throw new IllegalArgumentException("ExecutionAction.UNSET should never reach RubyContext");
-
+                        switch (getContext().getOptions().DEFAULT_EXECUTION_ACTION) {
+                            case NONE:
```

Comment:
Is it a bug to reach Truffle::Boot.main with no execute action? In what situation can it happen?

Reply:
When it's embedded you might want to do nothing until later, or when just showing the version (even though that's shortcutted in launcher).